subsys/usb/device/class/cdc_acm.c

```diff
@@ -621,13 +623,9 @@ static void cdc_acm_irq_tx_disable(const struct device *dev)
  */
 static int cdc_acm_irq_tx_ready(const struct device *dev)
 {
-	struct cdc_acm_dev_data_t * const dev_data = dev->data;
+	struct cdc_acm_dev_data_t *const data = dev->data;
 
-	if (dev_data->tx_irq_ena && dev_data->tx_ready) {
-		return 1;
-	}
-
-	return 0;
+	return data->tx_irq_ena && data->configured && data->tx_ready && !data->suspended;
```

Comment:
I do not think it is right, whether configured or not, UART device itself can fill data into FIFO, it will also break established expectations.

Reply:
I think it is right.

cdc_acm_fifo_fill() will not accept data. 

```
	if (!dev_data->configured || dev_data->suspended) {
		LOG_WRN("Device not configured or suspended, drop %d bytes",
			len);
		return 0;
	}
```

This cause an inifinte loop when using cdc_acm_irq_tx_ready() as in
zephyr/samples/subsys/usb/cdc_acm/src/main.c interrupt_handler() and many other projects.

Reply:
cdc_acm_irq_tx_ready() is implementation of uart_irq_tx_ready(), where the descriptions is "Check if UART TX buffer can accept a new char", `TX buffer`, it is not `Check if UART controller can transfer a new char`.

Reply:
Well... but it can't (or at least: won't). It will always return 0 because it will always take zero chars. That's not accepting a new char.

Reply:
The suspend logic in CDC ACM class implementation itself is not quite right.

Linux will suspend the ttyACM device if device claims to support remote wakeup and there is no data from host to device for some time. The device will be woken up by the host if and only if the host wants to send something. If the device has something to send to host, it is up to device to issue to remote wakeup - and it is what Linux expects the ttyACM device supporting remote wakeup will do.

Reply:
https://github.com/zephyrproject-rtos/zephyr/pull/73828/commits/490bd42693d5fdd2fa7bfd7c00c34fa076b05e4d#r1646202774 not addressed.